tests/chainerx_tests/unit_tests/routines_tests/test_manipulation.py

```diff
@@ -777,3 +784,160 @@ def forward_xp(self, inputs, xp):
 def test_expand_dims_invalid(xp, shape, axis):
     a = array_utils.create_dummy_ndarray(xp, shape, 'float32')
     return xp.expand_dims(a, axis)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Single axis
+    chainer.testing.product({
+        'shape': [(2, 2, 2)],
+        'axis': [*range(3)] + [*range(-1, -3, -1)],
+    })
+    + chainer.testing.product({
+        'shape': [(3, 3, 2, 3, 3)],
+        'axis': [*range(5)] + [*range(-1, -5, -1)],
+    })
+    + chainer.testing.product({
+        'shape': [(3, 0, 2, 0, 3)],
+        'axis': [*range(5)] + [*range(-1, -5, -1)],
+    })
+    + chainer.testing.product({
+        'shape': [(1, 2, 3, 1, 3, 3)],
+        'axis': [*range(6)] + [*range(-1, -6, -1)],
+    })
+    + chainer.testing.product({
+        'shape': [(3, 4, 5, 2, 3, 5)],
+        'axis': [*range(6)] + [*range(-1, -6, -1)],
+    })
+    + chainer.testing.product({
+        'shape': [(1,)],
+        'axis': [*range(1)] + [*range(-1, -1, -1)],
+    })
+    # Multiple axes
+    + chainer.testing.product({
+        'shape': [(1, 3, 4)],
+        'axis': powerset([*range(3)]) + powerset([*range(-1, -3, -1)]),
+    })
+    + chainer.testing.product({
+        'shape': [(3, 0, 2)],
+        'axis': powerset([*range(3)]) + powerset([*range(-1, -3, -1)]),
+    })
+    + chainer.testing.product({
+        'shape': [(1,)],
+        'axis': powerset([*range(1)]) + powerset([*range(-1, -1, -1)]),
+    })
+))
+@chainer.testing.parameterize_pytest('is_contiguous', [True, False])
+class TestFlip(op_utils.NumpyOpTest):
+
+    def setup(self, dtype):
+        # TO-DO(kshitij12345) : Remove when #6621 is in.
```

Comment:
A nitpick but `TODO` instead of `TO-DO`.

Reply:
Done.